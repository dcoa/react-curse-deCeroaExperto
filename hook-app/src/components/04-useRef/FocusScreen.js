import React, { useRef } from 'react'
import '../01-useState/counter.css'
// este es el caso basico pero no muy usado realmente
export const FocusScreen = () => {
    const focus = useRef();

    const handleClick = () => {
        focus.current.select()
    }
    return (
        <div>
            <input
             type="text"
             name="name"
             className="form-control"
             placeholder="Nombre"
             autoComplete="off"
             ref={focus}
             />
            <button className=' btn btn-outline-success mt-5' 
            onClick={handleClick}
            >Focus</button>
        </div>
    )
}
