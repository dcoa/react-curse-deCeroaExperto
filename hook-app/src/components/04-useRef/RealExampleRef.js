import React, { useState } from 'react'
import '../01-useState/counter.css'
import { MultipleCutomHooks } from '../03-examples/MultipleCutomHooks'

// este ejemplo el useRef se usa en el useFech que esta en el MultipleCutomHooks para evitar que haga el llamado a la api con el componete desmontado
export const RealExampleRef = () => {
    const [show, setShow] = useState(false);
    return (
        <div>
        <h1>useRef real use case</h1>  
         <hr/>
        <button className=' btn btn-outline-success mb-5' 
        onClick={() => setShow(!show)}
        >Show quotes</button>
         {show && <MultipleCutomHooks/>}

        </div>
    )
}
