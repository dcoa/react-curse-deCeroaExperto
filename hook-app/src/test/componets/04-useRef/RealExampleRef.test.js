import React from "react"
import { shallow } from "enzyme"
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef"

describe('Test on RealExampleRef', () => {
    const wapper = shallow(<RealExampleRef/>)

    test('should render the component', () => {
        expect(wapper).toMatchSnapshot();
        expect(wapper.find('MultipleCutomHooks').exists()).toBeFalsy();

    })
 
    test('should show the component MultipleCutomHooks', () => {
        wapper.find('button').simulate('click');
        expect(wapper).toMatchSnapshot();
        expect(wapper.find('MultipleCutomHooks').exists()).toBeTruthy();
        
    })
})
