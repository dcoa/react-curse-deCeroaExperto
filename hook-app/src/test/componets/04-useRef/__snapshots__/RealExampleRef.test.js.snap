// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on RealExampleRef should render the component 1`] = `
<div>
  <h1>
    useRef real use case
  </h1>
  <hr />
  <button
    className=" btn btn-outline-success mb-5"
    onClick={[Function]}
  >
    Show quotes
  </button>
</div>
`;

exports[`Test on RealExampleRef should show the component MultipleCutomHooks 1`] = `
<div>
  <h1>
    useRef real use case
  </h1>
  <hr />
  <button
    className=" btn btn-outline-success mb-5"
    onClick={[Function]}
  >
    Show quotes
  </button>
  <MultipleCutomHooks />
</div>
`;
